src/server/game/Spells/Spell.cpp

```diff
@@ -3622,6 +3622,8 @@ void Spell::_cast(bool skipCheck)
             SendCastResult(castResult);
             SendInterrupted(0);
 
+            sScriptMgr->OnSpellGo(m_caster->ToCreature(), this, false);
```

Comment:
for this hook no need 1-st arg, you can extract this with `GetCaster()->ToCreature()`

Reply:
remove unessecary 1st parameter of hook OnSpellGo